javacutil/src/org/checkerframework/javacutil/AnnotationUtils.java

```diff
@@ -98,12 +99,15 @@ public static boolean areSame(
             return false;
         }
 
-        Map<? extends ExecutableElement, ? extends AnnotationValue> elval1 =
-                getElementValuesWithDefaults(a1);
-        Map<? extends ExecutableElement, ? extends AnnotationValue> elval2 =
-                getElementValuesWithDefaults(a2);
+        // This commented implementation is less efficient.  It is also wrong:  it requires a
+        // particular order for fields, and it distinguishes the long constants "33" and "33L".
```

Comment:
How is the new code handling "33" and "33L" the same way? Wouldn't it reach the `Objects.equals(val1, val2)` and be different, if those two literals are in fact represented differently?

Reply:
The old code was treating `33` and `33L` differently, and the new code treats them as the same.  I don't have time to write a long comment about it, but this comment lets an interested reader do that investigation.